### ISSUE TITLE
Windows Credential : Piwigo infos not always retrieved.

### ISSUE THREAD

**Issue by XavierP56** [darktable-org/darktable#16603]
Start / Quit darktable 10 times in a row: At some point, you'll see sometimes the credentials are not retrieved in the Piwigo Export. It is a random issue. 

_Originally posted by @XavierP56 in https://github.com/darktable-org/darktable/issues/16593#issuecomment-2048233312_

**Comment by zisoft** [comment]
@XavierP56: With PR #16609 I have made some improvements, can you please test the next nightly build when this PR is merged?

In case of any errors, log outputs should be generated, please check with `-d pwstorage`.

**Comment by XavierP56** [comment]
Hello. I still see the issue randomly with darktable-4.7.0+912.gcc9415aadc-win64. I launch from cdm using darktable -d pwstorage but where can I find the log and what is the name of the file you are looking for ?

**Comment by zisoft** [comment]
On Windows it should be in `C:\User\<your username\AppData\Local\darktable\darktable.log`

I'm afraid there is not much I can do. The API is pretty much simple (just a "read credentials for a given name"). If that works well 9 of 10 times, what can we do?

**Comment by victoryforce** [comment]
> On Windows it should be in `C:\User\<your username\AppData\Local\darktable\darktable.log`

More precisely `%LOCALAPPDATA%\Microsoft\Windows\InetCache\darktable\darktable-log.txt`. You cannot navigate to this file in File Explorer because it is located in hidden directories. Just type or paste the path to it in the Explorer path field. Please attach this file to your comment.

**Comment by XavierP56** [comment]
I don't know the Darktable internals but it looks that at some point you get the answer in one thread and another thread initialize the widget to empty fields.  Depending of how the scheduling occurs, that might explain.
I have attached the log. Since the password was in cleartext (bad !), I have changed it :-)
Hope you'll find something.

[darktable-log.txt](https://github.com/darktable-org/darktable/files/14970832/darktable-log.txt)

**Comment by zisoft** [comment]
@XavierP56: I have now made some more improvements, please test with the next nightly build when #16620 is merged.

I have made countless restarts of darktable and cannot reproduce the problem. If the issue still exists there must be something different and I need a reproducible scenario.

**Comment by XavierP56** [comment]
Hi Mario.
Still does not work. Now it never works. See logs.
[darktable-log.txt](https://github.com/darktable-org/darktable/files/15000700/darktable-log.txt)

**Comment by zisoft** [comment]
Since I have changed the structure of the credential entry you need to fill in the credentials and hit "Login" to save.

Anyway, the error message tells "Entry not found" which I could never reproduce. So where is the difference?

**Comment by XavierP56** [comment]
I did log again of course. I was connected then I quit Darktable and started again.
I've just uninstalled previous version, nothing different...

**Comment by zisoft** [comment]
>  By the way, my Darktable is in French, I'm using a French Windows 11.

Maybe that's a hint! I will check in that direction.
Can you try with english locale?

**Comment by zisoft** [comment]
The name of the entry is `darktable - piwigo`. 

Can you please check with Control Panale -- Windows Credentials?

**Comment by XavierP56** [comment]
Putting Darktable in English locale makes no difference.
I have opened le Gestionnaire d'identification (Credential Manager) and... there is NO darktable - piwigo entry in it.
Which probably explains why it fails.

**Comment by XavierP56** [comment]
Before starting Darktable, this entry wasn't here.
I enter my credential, log on Piwigo and then quit.
And I see in Windows Credential this:
![image](https://github.com/darktable-org/darktable/assets/5458970/addeaab0-f89e-4582-8479-60f4dacca2fd)
Very strange. Seems you create the entry in the wrong UTF or something like that...

**Comment by zisoft** [comment]
Great hint, thank you!

I think I have finally fixed this.
Can you please download the Windows build from my darktable fork: https://github.com/zisoft/darktable/releases and see if it works now.

I everything is ok I will prepare a (hopefully last) PR for the Windows pwstorage backend.

Thanks for the tests and you patience!

**Comment by XavierP56** [comment]
Hi Mario,

I took darktable-4.7.0+939.gbafd9f2934-win64 from your link.
It still see the issue.
![image](https://github.com/darktable-org/darktable/assets/5458970/c73e8b0a-487f-4bc2-adfd-dab422657d27)

Do I have the right version ?

**Comment by zisoft** [comment]
Next try: https://github.com/zisoft/darktable/releases/download/nightly/darktable-4.7.0+940.g74b868ce71-win64.exe

**Comment by XavierP56** [comment]
It's a little better :-)
But not working totally !
![image](https://github.com/darktable-org/darktable/assets/5458970/fa65faae-9aec-4d4d-9c24-697748a4ab6a)
However, when I load Darktable, I have:
![image](https://github.com/darktable-org/darktable/assets/5458970/f5507f79-8647-4252-b79e-1a275a7d9d6d)

So it seems you save well but don't retrieve well :-)
We are on the right track !

**Comment by XavierP56** [comment]
And here is the log (I changed the passwd to ---)
[darktable-log.txt](https://github.com/darktable-org/darktable/files/15015559/darktable-log.txt)

**Comment by XavierP56** [comment]
The serveur should be imagenolffeenne.piwigo.com
The user Xavier and of course the password I keep for myself :-)

**Comment by XavierP56** [comment]
Here is the log.
[darktable-log.txt](https://github.com/darktable-org/darktable/files/15015670/darktable-log.txt)

The problem is probably here:
     0,1972 [pwstorage_new] using windows credentials backend for username/password storage.
     0,9299 [pwstorage_windows_credentials_get] reading (imagenolffeenne.da, {"server":"imagenolffeenne.da","username":"Xavier","password":"-------"})

server should be I think imagenolfeenne.piwigo.com

**Comment by zisoft** [comment]
In your log I see entries from 2024/04/14:

```
   301,4798 [pwstorage_windows_credentials_get] reading (https://imagenolffeenne.piwigo.com, {"server":"https://imagenolffeenne.piwigo.com","username":"Xavier","password":"*********"})
   301,4799 [pwstorage_windows_credentials_set] storing (https://imagenolffeenne.piwigo.com, {"server":"https://imagenolffeenne.piwigo.com","username":"Xavier","password":"*********"})
```

so writing and reading looks good.

The server name is taken from the input field in the export module. No modifications.

**Comment by XavierP56** [comment]
But look at my screenshot above : Once I quit and restart darktable, I don't find
imagenolffeenne.piwigo.com
Xavier
<Password>

So you might get them from Windows Credential, they don't appear properly in the dialog box. Something wrong.

**Comment by XavierP56** [comment]
Hi,

1) Destroyed entry in Windows Credential
2) Darktable in French locale
3) Launch Darktable
![image](https://github.com/darktable-org/darktable/assets/5458970/9f8b86bb-015f-45dd-8521-6b420e38fdb9)
4) Ctrl Q to quit Darktable
5) Start Darktable again
6) Look...
![image](https://github.com/darktable-org/darktable/assets/5458970/23195f9f-ca53-45f2-8757-83d72ff9b043)

Something is wrong.

**Comment by zisoft** [comment]
Hi Xavier, here the next attempt: https://github.com/zisoft/darktable/releases/download/nightly/darktable-4.7.0+953.ge626af0a33-win64.exe

I changed again the structure so please remove the credential entries first.

This is all poking around in the dark because I still cannot reproduce.

**Comment by XavierP56** [comment]
Hi Mario !

It works now !
Great job !

**Comment by flop25** [comment]
Hello
I came to look for another issue, I found this one, that I still experience
I use 4.8.1, on a French Win11
Windows Credential has nothing related to darktable or piwigo
I have never ever saw the credentials being registered since I've been using DT; I thought it was 'normal' until I read this thread

**Comment by zisoft** [comment]
@flop25: Did you enable the Windows Credentials Storage backend in the settings?

![Bildschirmfoto 2025-01-02 um 18 01 54](https://github.com/user-attachments/assets/4e7edea5-ed1c-43ec-bac8-f844ca2efdf0)

**Comment by flop25** [comment]
Thank you very much for the quick reply and screenshot 
It was set to None but it had the white round saying it has been modified from default, but I swear I never changed that (especially to none)!
I changed to windows credentials, I had to restart twice but now it seems to works, thank you again.

initially I went for an issue were sometimes it says it's impossible to get the album list of and I have to restart DT, so quite minor.
